src/goto-symex/symex_stack.cpp

```diff
@@ -0,0 +1,32 @@
+/*******************************************************************\
+
+Module: Symbolic Execution - Stack tracking
+
+Author: Rafael Sá Menezes
+
+Date: April 2020
+
+\*******************************************************************/
+
+#include <cassert>
+#include <goto-symex/goto_symex.h>
+#include <util/expr_util.h>
+#include <util/irep2.h>
+
+void goto_symext::process_stack_size(expr2tc &expr)
+{
+  code_decl2t &decl_code = to_code_decl2t(expr);
+  cur_state->top().frame_info.add_decl(decl_code);
+  if(stack_limit != -1)
+  {
+    static unsignedbv_type2tc u32type(32);
```

Comment:
Shall we set it to 32 or 64 according to the option given in the command-line?

Reply:
Why static? The `irep2tc`s are shared_pointers.

Also, you should use the `type_pool` methods to get default types. I suggest using `get_uint64` regardless of the arch.